### ISSUE TITLE
Add Support for trying this code out without vagrant

### ISSUE THREAD

**Issue by Jakhotiya** [magento-research/venia-pwa-concept#70]
This repository doesn't work well without vagrant. We need to add support for trying out this code without needing vagrant. On my local I had to fix many url problems to get venia work. 

I would like to help but can somebody point me in right direction

**Comment by DrewML** [comment]
Hi @Jakhotiya - appreciate the feedback. Can you outline what problems you had attempting to set this up without Vagrant?

The (admittedly sparse) setup docs go through the Vagrant setup process since that seems to be the most common among m2 users, but most of us on the team are using this with Valet+ on our hosts.

**Comment by Jakhotiya** [comment]
### Environment setup
magento 2.3-develop branch
nodejs v9
nginx php7.1 fpm mysql stack
store url on my local 23-develop.to (Couldn't use url of the dev server provided with this repo)

### Setting up the module and theme
First I downloaded the code from github and moved the module to app/code/Magento and venia theme
to app/design/frontend/Magento/. Then I ran npm install in theme directory to setup the dependencies. I ran npm start to start up the dev server. This generated client.js and other javascript files as expected. After stopping dev server I ran setup:upgrade to install the theme and the pwa module. I activated venia theme. After opening my store front these are issues I had to fix
1) **404 for client.js** (because the path was /pub/static/frontend/Magento/venia/en_US/client.js instead of /pub/static/frontend/Magento/venia/en_US/js/client.js)
2) client.js:1 Uncaught ReferenceError: webpackJsonp is not defined
    at client.js:1. To fix this I had to include vendor.js before client.js
3) Everything loads but screen remains black with "loading" text. I noticed response from http://23-develop.to/graphql an request was 
```
{"data":{"urlResolver":null}}
```
It would be great if we could move the module-pwa in a separate repository as pointed out in [issue](https://github.com/magento-research/venia-pwa-concept/issues/62)

I'm still working on the last issue. But overall I would love to streamline these things and document them.

**Comment by Jakhotiya** [comment]
Thanks @leon0v That does help in changing the response received from server. Now instead of null it is
```
{"data":{"urlResolver":{"type":"CMS_PAGE"}}}
```
But interestingly That did not change anything in the UI. I still see "Loading" text. Any thoughts on that?

**Comment by leon0v** [comment]
@Jakhotiya What does the console in your browser output?

**Comment by Jakhotiya** [comment]
Console says
Routing resolve failed
 TypeError: Failed to construct 'URL': Invalid base URL
I will try you fix.

**Comment by leon0v** [comment]
What parameters do you use when initializing a new Peregrine() in your index.js?

Mine are:

`const app = new Peregrine({
    apiBase: new URL('/graphql', location.origin).toString(),
    __tmp_webpack_public_path__: __webpack_public_path__ + "/en_US"
});`

**Comment by Jakhotiya** [comment]
My code used to match code base
```
const app = new Peregrine({
    apiBase: new URL('/graphql', location.origin).toString(),
    __tmp_webpack_public_path__: __webpack_public_path__
});
```

After reading you comment I tried 

```
const app = new Peregrine({
    apiBase: new URL('/graphql', location.origin).toString(),
    __tmp_webpack_public_path__: __webpack_public_path__ + "/en_US"
});
```
That changes error to Routing resolve failed
 Error: 404. Let me take a look at this and I should be able to come up with an answer. I would like to point out that I'm not using webpack development server. I'm using regular magento 2 url on port 80

**Comment by Jakhotiya** [comment]
Thank you very much @DrewML for the explanation and a hint. I got dev server to work. I had forgotten to change host url in .env and kept getting proxying error(cryptic). After running dev server I do not have to worry about vendor.js. Indeed it has nothing to do with vagrant.

I noticed even after having my dev server running  javascript file requests kept on going to 404. I looked in pub/static/frontend/Magento/venia/en_US/ but the file did not exist. I was confused, because I thought webpack-dev server would generate such an file named client.js . **Only after much digging did I realize that dev server will keep all assets in memory hence there would be no physical file called client.js** 

One reason for such confusion was `/pub/static/frontend/Magento/venia/en_US/client.js` file path in `root.phtml`. Regular magento dev would think that this is a mistake. Especially if you run a build, you notice that client.js is correctly generated at path `/pub/static/frontend/Magento/venia/en_US/js/client.js`. My question is could we configure dev server to serve `/pub/static/frontend/Magento/venia/en_US/js/client.js` instead of `/pub/static/frontend/Magento/venia/en_US/client.js` and If I create pull request for such an change would it be acceptable?
